### ISSUE TITLE
Add support for WeMo Dimmer Light Switch

### ISSUE THREAD

**Issue by mog1919** [eclipse/smarthome#3632]
The WeMo Dimmer Light Switch is not auto-detected as a "Thing" with in OpenHab with the WeMo binding.  

Also mentioned in this thread
https://community.openhab.org/t/wemo-dimmer-switch-not-auto-detected/29690

**Comment by hmerk** [comment]
As mentioned in the forum post, we need captured traffic between WeMo App and the dimmer devices, as there is no documentation of the used attributes to be found.

**Comment by hmerk** [comment]
Fist Binding version to test can be found at
[(https://drive.google.com/open?id=0BxdTuQI5IDvwTFppUUUzY0E0REk](url)
So far, the following should work:

- Autodiscovery
- Switching ON/OFF (channel "state")
- Polling ON/OFF state every 2 minutes.

**Comment by hmerk** [comment]
Making progress with the help of the community.
Main functionality is handled in the SetBinaryState/GetBinaryState methods of "basicevent" actionsService.

It also seems that subscribing to GENA events should be possible giving the following result:

`<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body>
	<u:GetBinaryStateResponse xmlns:u="urn:Belkin:service:basicevent:1">
		<BinaryState>0</BinaryState>
		<brightness>91</brightness>
		<fader>300:-1:1:0:87</fader>
		</u:GetBinaryStateResponse>
</s:Body> </s:Envelope>
`

**Comment by hmerk** [comment]
New version ready for testing.
(https://drive.google.com/open?id=0BxdTuQI5IDvwTFppUUUzY0E0REk

Changelog:
+ Added methods for channel brightness.

**Comment by mikenemat** [comment]
Hey there @hmerk . I'm testing your latest build with the Wemo Dimmer right now and the following things are working really well: Autodiscovery, polling, BinaryState events.

However, I can't seem to get OpenHAB to control the Wemo Dimmer. The channels/items are configured properly. The actions just do not seem to be sent over the network to the Dimmer. Changes to the dimmer via the Wemo app are reflected in OpenHAB, indicating it can receive data. It just doesn't appear to send.

**Comment by mikenemat** [comment]
Just noticed another issue. When the brightness levels are synchronized, the value appears to be incorrect. For example, <brightness>62</brightness> in the logs translates to 24% in PaperUI

**Comment by hmerk** [comment]
@mikenemat 
Thanks for your feedback, I will check your logs to see where the issue is.
According to brightness, that seems to bee an easy one. As for the bulbs, I recalculated the value, which seems to be unnecessary for the dimmer. Just a few lines of code to delete.

**Comment by hmerk** [comment]
Next version ready for testing.
https://drive.google.com/open?id=0BxdTuQI5IDvwTFppUUUzY0E0REk

Changelog:

- Removed recalculation of brightness values
- Corrections for command-handling
- Changed some logging settings for testing 

@mikenemat 
Please test again.

**Comment by seasickmonkey** [comment]
Bender here. I have installed the newest version and the dimmer works! However, the switch channel stopped working.

**Comment by mikenemat** [comment]
Seeing the same thing as @seasickmonkey. Dimmer works great (while turned on) but cannot switch on/off

**Comment by hmerk** [comment]
@seasickmonkey @mikenemat 
Thank you both for testing. I will check @seasickmonkey logs this evening.
As it worked before, it should be easy to fix.

**Comment by hmerk** [comment]
@seasickmonkey @mikenemat

Dumb me, it was just a one letter type, fixed it it now the state channel schould be back to work.
New version at the same link.

**Comment by mikenemat** [comment]
Working great @hmerk . Thanks!

I'm a bit new to OpenHAB 2 and channels. What is the proper way to control both state and brightness from a single item? 

Right now I have a rule like this...but it was never needed with OpenHAB 1.x style Dimmer items.

rule "wemoDimmer"
when
    Item wemoDimmerBrightnessItem received command
then
    if(wemoDimmerBrightnessItem.state == 0) wemoDimmerStateItem.sendCommand(OFF)
    if(wemoDimmerBrightnessItem.state >  0) wemoDimmerStateItem.sendCommand(ON)
end

**Comment by hmerk** [comment]
This is typically WeMo ... WeMo App also shows two "widgets", a switch and a slider, caus you can switch the lights/dimmer of without changing the brightness value. After switching on, the old value will be restored.
You can define a slider and a switch in your sitemap and bind both to channel brightness, it accepts ON/OFF commands as well, were ON is interpreted as 100% and OFF is 0%.

**Comment by mikenemat** [comment]
Thanks @hmerk that makes perfect sense. I will keep the rule in order to emulate traditional dimmers.

**Comment by seasickmonkey** [comment]
Thank you, @hmerk!

**Comment by hmerk** [comment]
@seasickmonkey @mikenemat 
Excellent guys, thanks for your feedback.
So as basics are up, we come to the fun stuff....
May I ask you to set the sleep timer through WeMo app to 26 minutes wait three minutes and enable the timer.
Wait another three minutes and disable the timer again.
This should result in three different log entries I ask you to post for analysis.
Do similar testing with the night mode, changing times, changing brightness, enable and disable nightmode.

Best
Hans-Jörg

**Comment by Bonanzarama** [comment]
@hmerk, I did the test for the timer (set to 26, 3minutes wait, enable, 3 minutes wait, disable.
My logs in /var/log/openhab2/ didn't show anything but I captured the output of log:display in karaf console if that's of help. 
[wemo_timer_test.txt](https://github.com/eclipse/smarthome/files/1100713/wemo_timer_test.txt)

**Comment by Bonanzarama** [comment]
Night mode: changed times (start 10pm, end 8:30am) and adjusted brightness, 3-4 min wait,  enable night mode (though it was only 7:30), wait ...10 minutes (or so, busy), disable, wait 3,  change start time to 7:30pm, wait 15 or 20, enable, wait 3, disable, wait.  Again, used log:display from karaf.

[wemo.night.test.txt](https://github.com/eclipse/smarthome/files/1100981/wemo.night.test.txt)

Apologies for excessive log info

**Comment by hmerk** [comment]
Thanks @Bonanzarama 
These logs are really helpfull.

**Comment by Bonanzarama** [comment]
I am a beginner (and not always observant enough) so I didn't notice the 'show more' text beside the channels in the PaperUI things.  Consequently, I thought to myself, "Maybe I need to uninstall/install so that the 'simple mode' of 'item linking' could generate the channels.  Anyhow, that's what I did: remove/re-add.

It added the dimmer with the 'switch' channel (wemo:dimmer:Dimmer-1_0-241717K1500766:state) but the channel that had previously been used for dimming (wemo:dimmer:Dimmer-1_0-241717K1500766:brightness) was missing.  And when I noticed the 'show more' it was not there either.  But all the channels for fader and nightmode are there! So I will play with those as I am able while I await advice (I'm not sure if this is an actual issue or my own inexperience).

**Comment by hmerk** [comment]
@Bonanzarama 
Sorry Brian, my mistake. When adding the new channels, I accidentily deleted the brightness channel.
I fixed this and uploaded a new version under the same link.

**Comment by Bonanzarama** [comment]
No trouble at all.  It's all a learning experience for me. Thanks for the fix.

**Comment by hmerk** [comment]
@Bonanzarama 
Any updates from testing ?

**Comment by Bonanzarama** [comment]
Update: 
I put basic config into the items config as follows at end. This displayed all controls as sliders.  Should I add more appropriate types (e.g., DateTime)?  I tried a few different types but then got nothing displayed: thought maybe I needed a restart of openhab but haven't tried yet. 

A comment, regarding the 'fader'... perhaps this should be called 'sleep timer' or 'sleep mode'.   
In the wemo app the "sleep" option  displays a message based on current duration settings:

"Lights would **be out** by xx:xx" when _Fade out_ is **OFF** 
or 
"Lights would **fade out** by xx:xx" when _Fade out_ is **ON**.

I'll try an figure out how to make things work in the next day or two; obviously (I hope it was obvious), the sliders are not showing state at all for me.

```
// from my .items file
// Fader : Lights fade out by specified time. 
// State reading fully implemented, switching working for "OFF" only atm.

Dimmer Wemo_Dimmer_Fader_State "State of Fader" <dimmable-30> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:faderState" }
Dimmer Wemo_Dimmer_Fader_Duration "Fader Duration" <clock> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:faderTime" }
Dimmer Wemo_Dimmer_Fader_Start "Time Fader Started" <clock> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:faderStart" }


// NightMode : Lights dim to preset amount during specified 'nightmode' hours 
// State reading implemented, no command handling in this version.

 Dimmer Wemo_Dimmer_Nightmode_State "Nightmode State" <moon> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:nightMode" } 
 Dimmer Wemo_Dimmer_Nightmode_StartTime "Start of Night" <clock> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:startTime" }
 Dimmer Wemo_Dimmer_Nightmode_EndTime "End of Night" <clock> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:endTime" }
 Dimmer Wemo_Dimmer_Nightmode_Brightness "Nightmode Brightness" <dimmablelight-100> { channel="wemo:dimmer:Dimmer-1_0-241717K1500766:nightModeBrightness" } 
```

**Comment by Bonanzarama** [comment]
I should note also that my lightbulbs are not truly appropriate for dimming. Consequently, I can't offer good info about how the wemo app would handle the dimming when 'fader' is enabled on the sleep timer. That will have to come when I find new bulbs.

**Comment by hmerk** [comment]
Thanks for your update Brian !
The channels are named afte the "original" methods or arguments of the wemo devices, therefore ist is called fader. In your UI, you can call it whatever you want :-)
Not all channels can be configured as slider. I prepared an update for the README as well
faderState -> Switch
faderTime -> Number or Dimmer, setpoint in UI
faderStart -> DateTime
nightMode -> Switch
startTime -> DateTime
endTime -> DateTime
nightModeBrightness -> Dimmer

I hope that helps.

**Comment by Bonanzarama** [comment]
With faderState as a Switch it does not seem to register any changes made in the wemo app.  I.e., whether I flip the fader switch to on/off or start/stop the sleep timer, no state change is apparent (I was monitoring in Basic UI and navigating to my dimmer from the BasicUI home screen. Similar behaviour with Nightmode.

HOWEVER, if I load the openHAB iOS app then it mirrors the BasicUI.  Flipping switches in either one will see the change reflected in the other (reloading the BasicUI web page where necessary). So that's something.
:)

**Comment by hmerk** [comment]
Sounds a bit stragen, could you provide the according logs.

**Comment by Bonanzarama** [comment]
Here is for the fader functionality.

The dimmer was off: no light on in the room.
In the wemo app, 'Fade out' was 'ON'.
At 5:30 I clicked 'start timer' in the wemo app.
The light turned on, in BasicUI the dimmer registered 'ON' and a short second or 4 later the fader registered 'ON'.
(sorry, obviously it's working somewhat).
I had the fader time set to 5min in wemo app so at 5:35 the light went out (after gradually dimming).
BasicUI showed power 'OFF' and still registered fader 'ON' (fader still 'ON' after 3 min)

Turned 'Fade Out' to 'OFF' in the wemo app. So it just will turn off when timer runs out (without fading out).
At 5:39:58(approximately) turned basicUI fader to off position
Clicked 'start timer' at 5:40
In BasicUI, power went 'ON'
Couple seconds later the fader also went on.
(success again!).

So obviously this is working.  Possibly my failure here had to do with a glitch in my hardware (had to power off at the breaker to even be able to turn on the light)... or more likely a glitch in the human running things.  My apologies.

At 5:45 light went 'OFF' in BasicUI/openHAB app and turned off.
the fader remains in 'ON' position in BasicUI/openHAB app.

It seems that the fader does not register the 'OFF' state when the timer has completed.

At 5:58 I turned off light and the 'fader' in openhab app.
Wemo app also now registers 'fade out' disabled and start timer 'off' (start timer correlates to the fader in openhab).

At 6:02 I turned on fader ('start timer') with wemo app.
At 6:04 I turned off fader ('cancel timer') with wemo app.

in BasicUI/openhab app light stays ON and fader stays ON. (and light physically stays on)

6:06 turned light off with wemo app.  Grabbed logs.
[Wemo.fader.log.txt](https://github.com/eclipse/smarthome/files/1120813/Wemo.fader.log.txt)

**Comment by txt8888** [comment]
All,  

I am very newbie to OpenHAB and home automation in general.

I stumble on HAB2 a couple weekend ago and managed to have it up and running on Pi with RazBerry z-wave controller.  Got a Z-wave dimmer to work.

in the midst of trying to see if HAB2 support WeMo Dimmer and found this article, it seemed a lot great works have been done and tested. 

And I like to see if there is any updates on this to the extend of on/off/dimmer-level? is this in the 2.1.0 release?

Thanks!

**Comment by hmerk** [comment]
No, there is no ETA, for two reasons.
First reason is a lack of time, second is that it is really difficult to implement devices that I don't have at hands. As I wrote ear

**Comment by angel12** [comment]
Hmerk- thanks for your work on this so far! I was wondering if you had the source code for your updated addon for the wemo dimmer posted somewhere? I looked on your github but couldn’t find it :(

**Comment by hmerk** [comment]
@angel12 so you did not look at the right place. It is in my branch 
wemo_fix3632.
Due to lack of time, there is no progress atm. I am still waiting to get my hands on a device.....

**Comment by hmerk** [comment]
@Bonanzarama 
I luckily could get a WeMo Dimmer finding it's way to germany. So there might be a good chance to continue development over Chritmas time ....
Will keep you updated here...

**Comment by Bonanzarama** [comment]
Good news! I had meant to get back to testing/logging for you but it seems I am unreliable. Have fun over Christmas.

**Comment by hmerk** [comment]
@Bonanzarama 
Next version ready for testing.
https://drive.google.com/open?id=0BxdTuQI5IDvwTFppUUUzY0E0REk

Working and tested functionality:
Switching ON/OFF
Setting Brightness
Switching NightMode ON/OFF
Changing FaderTime
Enable/Disable FaderMode
Start/Stop Timer

**Comment by juscruzin777** [comment]
Hello.  

Noob here. First post. I installed OH2 on my Win7 machine just a few days ago. I received my Wemo Dimmer in the mail this morning. I've been searching and searching on how to install the version provided by hmerk (thank you very much!) for testing.

Can anyone assist or point me in the right direction for more information in traveling this rough and rugged road?!

**Comment by hmerk** [comment]
Just drop the downloaded jar file into your addons folder.

**Comment by juscruzin777** [comment]
Wow, thank you for the quick reply!
I did as you instructed but cannot find any reference to the added file in Paper UI. 

Any suggestions?

**Comment by hmerk** [comment]
You should find it under config/bindings in PaperUI.
Also please check in Karaf console if bundle:list command shows it as running.

**Comment by juscruzin777** [comment]
I see your original Wemo Binding in PUI that I installed a few days ago, but do not see anything about "dimmers". I do not see the new binding listed in Karaf console after running bundle:list command.

**Comment by hmerk** [comment]
Ok, you have missed to uninstall WeMo Binding first and the clear /userdata/cache to remove the old Binding completely.
After a clean restart put the new binding into addons folder again.

**Comment by juscruzin777** [comment]
I uninstalled the Wemo Binding in PaperUI
Cleared /userdata/cache of all files and folders 
Restarted OpenHAB
Put the new binding into /addons
Installed Wemo Bindings

I still do not see any "dimmers". I only see the original Wemo Binding under Config/bindings in PaperUI.  
Will the dimmer be a separate binding or will it be listed under the original Wemo Binding with the coffee maker, switch... ?

**Comment by hmerk** [comment]
If you scan for new WeMo devices, it should appear in Inbox.
Could you post what Binding version is shown in Karaf console.
Another question, is your Dimmer working with original WeMo App?

**Comment by juscruzin777** [comment]
The original WeMo app is buggy and intermittent at best. WeMo hardware seems solid, but app is in need of big help.  OpenHAB straight away recognized my light switch and mini switch. I was even able to add rules to turn on and off at specific dates. OpenHAB is great!

Other than the initial discovery in the installation of WeMo app, I was never able to control the dimmer with WeMo app.

**Comment by juscruzin777** [comment]
When scanning for new WeMo devices, nothing is found.

I will try posting binding versions in Karaf...

**Comment by juscruzin777** [comment]
These are the only references I see to WeMo in Karaf bundle:list


193 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding
194 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome Wemo Binding
195 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome UPnP Transpor
t Bundle
openhab>

**Comment by hmerk** [comment]
Rhis clearly says your old Wemo Binding is active, not the new one.
So again, please stop OH, delete content of cache and temp folder.
If you have added Wemo Binding to addons.cfg, please remove that entry there and restart OH
Check in console if you have just one Entry for Wemo Binding and that‘s not the 0.9.0.b5 one.

**Comment by juscruzin777** [comment]
Still unable to see "dimmers" in PaperUI

I stopped OH and deleted contents of /openhab2/cache/
No WeMo bindings in addons.cfg

console show no instance of wemo bindings

193 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding
194 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome Wemo Binding
195 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome UPnP Transpor
t Bundle
openhab>
 
But when I open PaperUI and addons>bindings>wemo bindings (2.1.0) and install, console shows this

193 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding
194 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome Wemo Binding
195 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome UPnP Transpor
t Bundle
openhab>

I placed the newly downloaded binding into /openhab2/addons
Still unable to see "dimmers" in PaperUI

**Comment by hmerk** [comment]
Ok, i forgot the obvious:
1. Uninstall Wemo Binding from PaperUi
2. Stop openHAB
3. Delete Wemo Binding in addons folder
4. Delete content of cache AND temp folder
5. Start openHAB
6. Check with console that there is no Wemo Binding installed
7. Place downloaded jar into addons folder
8. Check in console if you got the new binding active

**Comment by juscruzin777** [comment]
hmerk,
No dimmer control yet, but making progress! Thank you for your time and patience!
It looks like I did not delete the TMP folder contents.

Doing steps 1 - 8 above, console now shows:
s: 188
191 | Active    |  80 | 2.1.0                  | HABPanel User Interface
192 | Resolved  |  75 | 2.1.0                  | openHAB Paper UI Theme Fragment
, Hosts: 189
193 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding
openhab>

After step 8 above, I did the following:
PaperUI  Add-ons>bindings>WeMo Bindings 2.1.0\Install

OH automatically searched and discovered my Wemo Light Switch and Mini Switch. 
No dimmer switch so I initiated another search, still no Wemo dimmer discovered.

I went back to console and checked what Wemo bindings are installed/active. It looks like the old Wemo binding came back up after going back into PaperUI and re-installing the Wemo bindings.

191 | Active    |  80 | 2.1.0                  | HABPanel User Interface
192 | Resolved  |  75 | 2.1.0                  | openHAB Paper UI Theme Fragment
, Hosts: 189
193 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding
194 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome Wemo Binding
195 | Active    |  80 | 0.9.0.b5               | Eclipse SmartHome UPnP Transpor
t Bundle
openhab>

I feel we're getting close! Any suggestions?

**Comment by hmerk** [comment]
No, don‘t install WeMo Binding through Paper UI again.
Jusz check in console if the new binding comes active. If it stays installed, please post the according logs here.

**Comment by juscruzin777** [comment]
Ok, I'm back to step 8.

console
s: 188
191 | Active    |  80 | 2.1.0                  | HABPanel User Interface
192 | Resolved  |  75 | 2.1.0                  | openHAB Paper UI Theme Fragment
, Hosts: 189
193 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding
openhab>


openhab.log
2017-12-23 10:52:18.241 [INFO ] [basic.internal.servlet.WebAppServlet] - Stopped Basic UI
2017-12-23 10:53:35.429 [INFO ] [.internal.GenericScriptEngineFactory] - Activated scripting support for ECMAScript
2017-12-23 10:53:39.459 [INFO ] [.dashboard.internal.DashboardService] - Started dashboard at http://192.168.27.134:8080
2017-12-23 10:53:39.659 [INFO ] [.dashboard.internal.DashboardService] - Started dashboard at https://192.168.27.134:8443
2017-12-23 10:53:40.006 [INFO ] [basic.internal.servlet.WebAppServlet] - Started Basic UI at /basicui/app
2017-12-23 10:53:40.016 [INFO ] [arthome.ui.paper.internal.PaperUIApp] - Started Paper UI at /paperui
2017-12-23 10:53:40.030 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2017-12-23 10:53:40.035 [ERROR] [org.eclipse.smarthome.binding.wemo  ] - FrameworkEvent ERROR - org.eclipse.smarthome.binding.wemo
org.osgi.framework.BundleException: Could not resolve module: org.eclipse.smarthome.binding.wemo [193]
  Unresolved requirement: Import-Package: org.eclipse.smarthome.io.transport.upnp

	at org.eclipse.osgi.container.Module.start(Module.java:434)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1582)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1562)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1533)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1476)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]

**Comment by hmerk** [comment]
Ok, that looks good. The Binding is not startet, as the UPnP Module is not installe when dropped into addons folder. Easiest solution, just install Sonos Binding through PaperUI even you don‘t use it.
Afte another restart, you should see Wmo Binding active

**Comment by juscruzin777** [comment]
I SEE WEMO DIMMER IN MY IN BOX!!
THANK YOU, THANK YOU, THANK YOU!

Thank you again for your time and patience.  I am excited about OpenHAB and the awesome support that follows.

**Comment by hmerk** [comment]
Glad you got it running. If you see any issues, feel free to post your comments here.
Mary X-MAS

**Comment by juscruzin777** [comment]
Will do, I will post any dimmer issues here... I have other Wemo issues that I will try and tackle myself first. If I have no luck, I will open a new topic.

And yes, MERRY CHRISTMAS!

**Comment by txt8888** [comment]
I also just received the Wemo Dimmer. I put the downloaded jar into: `/usr/share/openhab2/addons`
from karaf console, I see:

`194 | Installed |  80 | 0.9.0.201712142306     | Eclipse SmartHome Wemo Binding`
when I tried to start this with the command `bundle:start 194` and I got this error

```
openhab> bundle:start -t 194
Error executing command: Error executing command on bundles:
	Error starting bundle 194: Could not resolve module: org.eclipse.smarthome.binding.wemo [194]
  Unresolved requirement: Import-Package: org.eclipse.smarthome.io.transport.upnp
```
So I install SONOS on PaperUI as suggested,  and rerun the `bundle:start -t 194` and it work.

Do I install the 
```
Wemo Binding
binding-wemo - 2.1.0
```
from PaperUI?  I just want to make sure before I click "`install`"

Thanks

**Comment by hmerk** [comment]
No, if you see Bundle 194 schown as active, all should be fine.
Check your inbox, it should list your WeMo Dimmer if it is connected to the same network as your openHAB Server. If it does not sho up, Start a scan for WeMo devices in Inbox.

**Comment by txt8888** [comment]
Thanks HMerk for quick replied.    I have not installed the Dimmer.  I will do shortly.   After I install, do I need to use Wemo App to do anything?  or just straight to OpenHAB2 ??  many thanks.  so happy to have this new toy for the holiday!

**Comment by hmerk** [comment]
After physical installation, you need the original WeMo App to let the Dimmer join your WIFI network.
This step cannot be done trough openHAB.

**Comment by txt8888** [comment]
I got it to work.   thank you for the great works on this Wemo dimmer.

**Comment by ianmaddocks** [comment]
@hmerk I am looking to get my ESP8266 mains dimmer to emulate a Wemo dimmer switch so I can control it from Alexa and OpenHAB. Is there any explanation of the Wemo dimmer’s discovery and action command/responses so can extended the fauxmo module to emulate a dimmer too?

Ideally I’d emulate the dimmer switch in my mains dimmer’s ESP8266 or I could use a RPi as a bridge.

Thanks in advance.   

Dimmer: https://www.tindie.com/products/nEXT_EVO1/ac-mains-dimmer-mpdmv75/

**Comment by hmerk** [comment]
@ianmaddocks 
There is no official documentation published by Belkin. All I have achieved is done by trial and error and packet sniffing. I even don‘t think that it is a good idea to emulate a WeMo Dimmer, cause it has some very specific functions like NightMode or SleepTimer with fade to Zero.
So the best way would be to implement a MQTT protocol with your basic dimming functions.

**Comment by ianmaddocks** [comment]
Thanks HMerk for the quick reply.

**Comment by hmerk** [comment]
The dimmer switch is still in experimental state, so don‘t expect correct behaviour.
I have very limited time atm, so don‘t know when it will be ready to be released.

**Comment by meatheadmike** [comment]
Do you have a link to your source? I couldn't find wemo_fix3632 under your github... Thanks!
